ranch/ranch.go

```diff
@@ -299,7 +299,7 @@ func (r *Ranch) Release(name, dest, owner string) error {
 			logrus.WithError(err).Errorf("unable to release resource %s", name)
 			return &ResourceNotFound{name}
 		}
-		if owner != res.Status.Owner {
+		if owner != "" && owner != res.Status.Owner {
```

Comment:
I think this is the wrong fix. if `owner` is empty this function wouldn't even be called, it would return at https://github.com/kubernetes-sigs/boskos/blob/b9220546cd7b5958c68f907035cfe18dd2182cfe/handlers/handlers.go#L136 and the error code should be 400 instead of 401 as seen in the Slack thread.

Reply:
That is correct. My only explanation is that the GetResource call to Kube API Server is returning bad data.

Reply:
I'm getting errors with the release handler not acquire handler.

https://grafana.knative.dev/d/BcbddCZVk/boskos-server-dashboard?orgId=1&from=now-6h&to=now

401s are coming from /release and /update

Reply:
this function `r.Release` is referenced at 3 different places in this codebase, and the `owner == ""` check exists at all 3 places. So I don't think this change has any effect as `owner` could never be empty here